docs/.vitepress/vitepress/components/navbar/vp-search.vue

```diff
@@ -52,6 +54,13 @@ function update(options: any) {
   }
 }
 
+function convertUrlToCurrentLang(url: string) {
+  const arr = url.split('/')
+  return langs.value.includes(arr[1])
+    ? `/${lang.value}/${arr.slice(2).join('/')}`
+    : url
+}
```

Comment:
I noticed that this is just rewriting the url.

Although it is not a problem to convert the url to an array and then convert it to a string, it is not an elegant solution.

Would you like to try to complete it by processing strings only? (eg. replace)

Reply:
I also think it's not elegant enough, but I'm not sure if the first part of the URL path is a valid lang.

So I need determine whether it is a valid lang first.

Reply:
You can match it through RegExp (eg.  `/^\/(langA|langB|langC)\b/`)

Reply:
hi, I have rewrited it